src/core/database.py

```diff
@@ -0,0 +1,78 @@
+# coding: utf8
+
+__contact__ = "info@hytech-imaging.fr"
+__copyright__ = "Copyright (c) 2021 Hytech Imaging"
+
+import os.path
+
+from PyQt5.QtWidgets import QMessageBox
+from qgis._core import (
+    QgsWkbTypes,
+    QgsFields,
+    QgsVectorFileWriter,
+    QgsCoordinateReferenceSystem,
+    QgsCoordinateTransformContext,
+    QgsFeatureSink,
+    QgsVectorLayer,
+    QgsProject,
+)
+
+CONST_DB_NAME = "sammo-boat.gpkg"
+CONST_LAYER_NAME = "session datas"
+
+class SammoDataBase:
+    def isDataBaseAvailableInThisDirectory(self, directory):
+        return os.path.isfile(self._getPathToDataBase(directory))
+
+    def getDbName(self):
+        return CONST_DB_NAME
+
+    def createEmptyDataBase(self, directory):
+        geom = QgsWkbTypes.Point
+        tableName = "emptyTable"
+        db = self._getPathToDataBase(directory)
+
+        fields = QgsFields()
+
+        opts = QgsVectorFileWriter.SaveVectorOptions()
+        opts.driverName = "GPKG"
+        opts.layerName = tableName
+
+        if not os.path.isfile(db):
+            opts.actionOnExistingFile = \
+                QgsVectorFileWriter.CreateOrOverwriteFile
+        else:
+            opts.actionOnExistingFile = \
+                QgsVectorFileWriter.CreateOrOverwriteLayer
+
+        crs = QgsCoordinateReferenceSystem.fromEpsgId(4326)
+
+        QgsVectorFileWriter.create(
+            db,
+            fields,
+            geom,
+            crs,
+            QgsCoordinateTransformContext(),
+            opts,
+            QgsFeatureSink.SinkFlags(),
+            None,
+            tableName,
+        )
+
+    def loadDataBase(self, directory):
+        db = self._getPathToDataBase(directory)
+        vlayer = QgsVectorLayer(db, self._getLayerName())
+        if not vlayer.isValid():
+            QMessageBox.critical(
+                None,
+                "Sammo-Boat plugin",
+                "Impossible to read the file " + self._getPathToDataBase(),
+            )
+        else:
+            QgsProject.instance().addMapLayer(vlayer)
+
+    def _getPathToDataBase(self, directory):
```

Comment:
property

Reply:
In this case, it is not really a getter because of the directory argument. But, as name convention, you prefer no to have the "get" prefix ?

Reply:
Ah yes indeed! But I don't like to pass around the `directory` everywhere. I'll take a look later to that particular point.

In the meantime, +1 to remove the `get` prefix :+1:

Reply:
Done commit #8 (33fc7e1847da170fdcc5362c7978609aace3250c)